File: src/relays.cpp
#include <chrono>
#include <csignal>
#include <iostream>
#include <thread>

#include "./simulation.h"

using namespace Relays;

bool _quit = false;

void leave(int /*signal_value*/)
{
    _quit = true;
}

int main(int /*argc*/, char** /*argv*/)
{
    struct sigaction signals;
    signals.sa_handler = leave;
    signals.sa_flags = 0;
    sigaction(SIGINT, &signals, nullptr);
    sigaction(SIGTERM, &signals, nullptr);

    std::cout << "Running a default (empty...) circuit\n";

    auto simulation = Simulation();
    simulation.run();

    while (!_quit)
        std::this_thread::sleep_for(std::chrono::milliseconds(50));

    std::cout << "Stopping the simulation...\n";
    simulation.stop();
    std::cout << "Simulation stopped\n";

    return 0;
}

File: tools/relays.py
#!/usr/bin/env python3

import signal

from pyrelays import Simulation
from time import sleep

exit: bool = False


def sig_handler(signo, stack_frame):
    global exit
    exit = True


if __name__ == "__main__":
    signal.signal(signal.SIGTERM, sig_handler)
    signal.signal(signal.SIGINT, sig_handler)

    simulation: Simulation = Simulation()
    print("Running a default (empty...) circuit")
    simulation.run()

    while exit is not True:
        sleep(0.1)

    print("Stopping the simulation...")
    simulation.stop()
    print("Simulation stopped")

